### PR TITLE
Calculate nvi for AcademicChapter

### PR DESCRIPTION
**Pull request by monaullah**
Test nvi calculation for academic chapter. Modified nviTypeQuery.sparql to select publication channel levels for academic chapter as well.

### CODE REVIEW

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 1]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
...
-    OPTIONAL { ?publication nva:entityDescription/nva:publicationDate/nva:year ?publicationYear . }
-
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:publisher/nva:level ?publisherLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:series/nva:level ?seriesLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:level ?journalLevel .  }
+    OPTIONAL {
+      SELECT distinct ?publicationYear WHERE {
+        ?publicationDate rdf:type ?type .
+        ?publicationDate nva:year ?publicationYear
+        FILTER(?type = <https://nva.sikt.no/ontology/publication#PublicationDate>)
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:20]: Replace `<https://nva.sikt.no/ontology/publication#PublicationDate>` with `nva:PublicationDate`

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 2]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
...
 
-    OPTIONAL { ?publication nva:entityDescription/nva:publicationDate/nva:year ?publicationYear . }
-
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:publisher/nva:level ?publisherLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:series/nva:level ?seriesLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:level ?journalLevel .  }
+    OPTIONAL {
+      SELECT distinct ?publicationYear WHERE {
+        ?publicationDate rdf:type ?type .
+        ?publicationDate nva:year ?publicationYear
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:19]: Prettier if:

```
        ?publicationDate rdf:type ?type ;
                         nva:year ?publicationYear .
```

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 3]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
...
+      }
+    }
+
+    OPTIONAL {
+      SELECT distinct ?publisherLevel WHERE {
+        ?publisher rdf:type ?type .
+        ?publisher nva:level ?publisherLevel
+        FILTER(?type = <https://nva.sikt.no/ontology/publication#Publisher>)
+      }
+    }
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:30]:
```
    OPTIONAL {
      SELECT distinct ?publisherLevel WHERE {
        ?publisher rdf:type ?type ;
                   nva:level ?publisherLevel .
        FILTER(?type = nva:Publisher)
      }
    }
```

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 4]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
...
+
+    OPTIONAL {
+      SELECT distinct ?seriesLevel WHERE {
+        ?publisher rdf:type ?type .
+        ?publisher nva:level ?seriesLevel
+        FILTER(?type = <https://nva.sikt.no/ontology/publication#Series>)
+      }
+    }
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:38]:
```
    OPTIONAL {
      SELECT distinct ?seriesLevel WHERE {
        ?publisher rdf:type ?type ;
                   nva:level ?seriesLevel .
        FILTER(?type = nva:Series)
      }
    }
```

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 5]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
...
+
+    OPTIONAL {
+      SELECT distinct ?journalLevel WHERE {
+            ?publisher rdf:type ?type .
+            ?publisher nva:level ?journalLevel
+            FILTER(?type = <https://nva.sikt.no/ontology/publication#Journal>)
+      }
+    }
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:46]:
```

    OPTIONAL {
      SELECT distinct ?journalLevel WHERE {
            ?publisher rdf:type ?type ;
                       nva:level ?journalLevel .
            FILTER(?type = nva:Journal)
      }
    }
```

File: expansion/src/main/resources/nviTypeQuery.sparql [hunk 6]
```diff
@@ -13,11 +13,37 @@ CONSTRUCT {
     ?publication nva:status ?publicationStatus .
     ?publication nva:entityDescription/nva:reference/nva:publicationInstance/rdf:type ?instanceType .
 
-    OPTIONAL { ?publication nva:entityDescription/nva:publicationDate/nva:year ?publicationYear . }
-
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:publisher/nva:level ?publisherLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:series/nva:level ?seriesLevel . }
-    OPTIONAL { ?publication nva:entityDescription/nva:reference/nva:publicationContext/nva:level ?journalLevel .  }
+    OPTIONAL {
```
Comment (brinxmat) [inline expansion/src/main/resources/nviTypeQuery.sparql:16]: So it's said, we know that the limits of the graph don't include > 1 occurrence of these, so I guess these are fine. In truth, a check should be included to ensure we don't get weird results from e.g. artistic subtypes that include a publishing date (not sure if this is relevant any more…) I'm also pretty sure that these cases aren't NVI-data anyway…

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: expansion/src/main/resources/nviTypeQuery.sparql (hunk 6 above)
Comment (monaullah) [reply expansion/src/main/resources/nviTypeQuery.sparql:16]: I agree, we can refactor these to make it more robust in the nva-nvi application

**Pull request merged**